pkg/host/internal/network/network.go

```diff
@@ -175,16 +179,42 @@ func (n *network) SetNetdevMTU(pciAddr string, mtu int) error {
 
 func (n *network) GetNetDevMac(ifaceName string) string {
 	log.Log.V(2).Info("GetNetDevMac(): get Mac", "device", ifaceName)
-	macFilePath := filepath.Join(vars.FilesystemRoot, consts.SysClassNet, ifaceName, "address")
-	data, err := os.ReadFile(macFilePath)
+	addressFilePath := filepath.Join(vars.FilesystemRoot, consts.SysClassNet, ifaceName, "address")
+	data, err := os.ReadFile(addressFilePath)
 	if err != nil {
-		log.Log.Error(err, "GetNetDevMac(): fail to read Mac file", "path", macFilePath)
+		log.Log.Error(err, "readInterfaceAddress(): fail to read address file", "path", addressFilePath)
 		return ""
 	}
 
 	return strings.TrimSpace(string(data))
 }
 
+func (n *network) GetNetDevNodeGUID(pciAddr string) string {
+	log.Log.V(2).Info("GetNetDevNodeGUID(): get node GUID", "pciAddr", pciAddr)
+	if len(pciAddr) > 0 {
+		rdmaDevicesPath := filepath.Join(vars.FilesystemRoot, consts.SysBusPciDevices, pciAddr, "infiniband")
```

Comment:
will this file exist even if we move the network device into the pod namespace?

Reply:
I believe this depends on the RDMA mode we have configured on the system. Take a look at [this](https://man7.org/linux/man-pages/man8/rdma-system.8.html#:~:text=NEWMODE%20%2D%20specifies%20the,can%20be%0A%20%20%20%20%20%20%20used.) doc.

There is a case where the file won't exist. If it doesn't exist, then we return empty string. Based on this comment https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/587#discussion_r1477863309, I changed the implementation of the check to take action only when the GUID (and MAC address in other cases) can be read. Otherwise do nothing. It won't catch 100% of manual actions done by the user on the system, but should be good enough.

WDYT?